File: 프로그래머스/lv2/17680. ［1차］ 캐시/［1차］ 캐시.py
def solution(cacheSize, cities):
    cache = []
    time = 0
    if cacheSize == 0:
        time = len(cities) * 5
    else:
        for i in cities:
            if len(cache) < cacheSize:
                if i.lower() in cache:
                    cache.remove(i.lower())
                    cache.append(i.lower())
                    time += 1
                else:
                    cache.append(i.lower())
                    time += 5
            elif i.lower() in cache:
                time += 1
                cache.remove(i.lower())
                cache.append(i.lower())
            else:
                time += 5
                cache.remove(cache[0])
                cache.append(i.lower())
    print(time)
    return time

File: 프로그래머스/2/17680. ［1차］ 캐시/［1차］ 캐시.java
import java.util.*;

class Solution {
    public int solution(int cacheSize, String[] cities){
        if(cacheSize == 0) return cities.length * 5;
        
        int answer = 0;
        Queue<String> cache = new LinkedList<>();
        
        for(String c : cities){
            String cl = c.toLowerCase();
            if(cache.contains(cl)){
                cache.offer(cl);
                cache.remove(cl);
                answer += 1;
            } else if(cache.size() < cacheSize){
                cache.offer(cl);
                answer += 5;
            } else if(cache.size() == cacheSize){
                cache.poll();
                cache.offer(cl);
                answer += 5;
            }
        }
        return answer;
    }
}